### ISSUE TITLE
Add `emplace` to AssociativeVector

### ISSUE THREAD

**Issue by drug007** [OpenXRay/xray-16#256]
and use it in `game_state_accumulator_state_register.cpp`

**Comment by dino8890** [comment]
Hi, I'm willing to take a stab at this, but after looking at the code, I'm not sure what is this supposed to achieve. How would `emplace` method differ from the already implemented `insert` method?

This is from the game_state_accumulator_state_register.cpp:

```
#define INIT_ACCUMULATIVE_VALUE(Kind, SomeClass) \
    m_accumulative_values.insert(std::make_pair(Kind, xr_new<SomeClass>(this)));
    
    // couple of lines below
    
    INIT_ACCUMULATIVE_VALUE(acpv_mad, player_state_mad);
```

**Comment by Xottab-DUTY** [comment]
The change is – removed `std::make_pair` call. `emplace` constructs objects in place.
```cpp
#define INIT_ACCUMULATIVE_VALUE(Kind, SomeClass) m_accumulative_values.emplace(Kind, xr_new<SomeClass>(this));
```

**Comment by dino8890** [comment]
Can you check out related pull request and see if this is what you had in mind? If so, I will replace calls to `insert` with calls to `emplace` wherever possible and let you so it can be merged.

**Comment by Xottab-DUTY** [comment]
Closed with #924, thanks!
